python/graphframes/graphframe.py

```diff
@@ -205,6 +205,44 @@ def bfs(self, fromExpr, toExpr, edgeFilter=None, maxPathLength=10):
         jdf = builder.run()
         return DataFrame(jdf, self._sqlContext)
 
+    def aggregateMessages(self, aggCol, msgToSrc=None, msgToDst=None):
+        """
+        Aggregates messages from the neighbours.
+
+        When specifying the messages and aggregation function, the user may reference columns using
+        the static methods in :class:`AggregateMessages`.
+
+        See Scala documentation for more details.
+
+        :param aggCol: the requested aggregation output either as
+            `pyspark.sql.Column` or SQL expression string
+        :param msgToSrc: message sent to the source vertex of each triplet either as
+            `pyspark.sql.Column` or SQL expression string (default: None)
+        :param msgToDst: message sent to the destination vertex of each triplet either as
+            `pyspark.sql.Column` or SQL expression string (default: None)
+
+        :return: DataFrame with columns for the vertex ID and the resulting aggregated message
+        """
+        # Check that either msgToSrc, msgToDst, or both are provided
+        if msgToSrc is None and msgToDst is None:
+            raise ValueError("Either `msgToSrc`, `msgToDst`, or both have to be provided")
+        builder = self._jvm_graph.aggregateMessages()
+        if msgToSrc is not None:
+            if isinstance(msgToSrc, Column):
+                builder.sendToSrc(msgToSrc._jc)
+            else:
```

Comment:
Oh I missed that earlier.  In Spark, we generally do this: https://github.com/apache/spark/blob/e3d2022e4b73c5bc352eb001ed334d91ce36109c/python/pyspark/ml/feature.py#L18

and: https://github.com/apache/spark/blob/e3d2022e4b73c5bc352eb001ed334d91ce36109c/python/pyspark/ml/feature.py#L2833

Could you do that approach please?